### ISSUE TITLE
Allow for configuration of MAX_CONNECTION_AGE and MAX_CONNECTION_AGE_GRACE for gRPC server

### ISSUE THREAD

**Issue by jaygazula27** [quarkusio/quarkus#41263]
### Description

I would like to be able to configure these options documented in gRPC's keepalive [docs](https://grpc.io/docs/guides/keepalive/#keepalive-configuration-specification):

- MAX_CONNECTION_AGE -> Maximum time that a channel may exist.
- MAX_CONNECTION_AGE_GRACE -> Grace period after the channel reaches its max age.

These properties can be set via the `ServerBuilder` class.

Use case:
As seen in the discussion [here](https://github.com/grpc/grpc/issues/12295#issuecomment-382794204), this appears to be the recommendation for allowing clients to recognize new server instances. This is especially relevant in Kubernetes where the number of server pods may increase, but the gRPC client will have no idea about them because of the nature of the long lived HTTP/2 TCP connection. This happens regardless of the load balancing strategy. More info can be found [here](https://kubernetes.io/blog/2018/11/07/grpc-load-balancing-on-kubernetes-without-tears/).

### Implementation ideas

Ideally, I would like to be able to configure these via application properties as such:

- `quarkus.grpc.server.max-connection-age`
- `quarkus.grpc.server.max-connection-age-grace`

This seems like a fairly simple change in `GrpcServerRecorder#buildServer`? If so, I'm happy to contribute via a PR.

**Comment by cescoffier** [comment]
I'm not sure the Vert.x server have these options? As we are more and more recommending this server, it would make sense to add these options there too.

**Comment by alesj** [comment]
@vietj ^^

**Comment by cescoffier** [comment]
@jaygazula27 Sorry to come back to this so late. It totally went of my radar. 

It is only supported by the _classic_ server. However, we can still support these properties and say in the documentation that is requires `use-separate-server=true`. 

Fancy to open a PR? @alesj and I would help you to get it in.

**Comment by jaygazula27** [comment]
@cescoffier No worries.
Is there any recommendation on what to do when using the new server? I'm concerned that scaling out the Quarkus app pods won't make a difference if the managed channel is cached by the client. Do you recommend that I ask in https://github.com/eclipse-vertx/vertx-grpc?

And regarding the PR, sure, I can take a crack at it even if it is just for the classic server!

**Comment by cescoffier** [comment]
@jaygazula27 Yes, that's the right place to post questions about the Vert.x server.

**Comment by cescoffier** [comment]
@jaygazula27 Still fancy a PR? @alesj  and I can help you.

**Comment by cescoffier** [comment]
It should be doable with Vert.x 5 .

**Comment by huytran0605** [comment]
Hi,  I’d appreciate it if you could update me on the status of this config

**Comment by cescoffier** [comment]
It requires Vertx 5, planned for 2026.
